File: ingeneo_technical_test/ingeneo_technical_test/src/main/java/com/example/ingeneo_technical_test/entity/dto/ProductDTO.java
/**
 * 
 */
package com.example.ingeneo_technical_test.entity.dto;

import java.util.Date;

import com.example.ingeneo_technical_test.enumerations.ProductType;
import com.fasterxml.jackson.annotation.JsonInclude;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Johan Casagua
 */
@Data @AllArgsConstructor @NoArgsConstructor @Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ProductDTO {
	private Long id;
	private String name;
	private int quantity;
	private String description;
	private float price;
	private ProductType type;
	private Date creationDate;
}

File: frontend/ingeneott/src/app/models/product.dto.ts
export class ProductDTO {
    id: number;
    name: string;
    quantity: number;
    description: String;
    price: number;
    type: ProductType;
    date: Date;
  }
  
  export enum ProductType {
    VEHICLES = 'VEHICLES',
    TECHNOLOGY = 'TECHNOLOGY',
    HOME_AND_FURNITURE = 'HOME_AND_FURNITURE',
    FASHION = 'FASHION',
    HEALTH_AND_MEDICAL_EQUIPMENT = 'HEALTH_AND_MEDICAL_EQUIPMENT'
  }